enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/grpc/ExtAuthService.java

```diff
@@ -149,7 +148,7 @@ private CheckResponse buildResponse(CheckRequest request, ResponseObject respons
             // query parameter. In this scenario, apiKey query parameter is sent within the property called
             // 'queryParamsToRemove' so that the custom filters also can utilize the method.
             if (responseObject.getQueryParamsToRemove().size() > 0) {
-                String constructedPath = constructQueryParamString(responseObject.getRequestPath(),
+                String constructedPath = RequestUtils.constructQueryParamString(responseObject.getRequestPath(),
```

Comment:
We could use the same pattern to add that dynamic metadata in buildResponse method method. Build response method is the place we always build the check response.